### PR TITLE
Plane: Add airbrake function for autoland

### PR DESCRIPTION
**Pull request by samuelctabor**
This adds an airbrake servo output function that is driven by the negative throttle values. This allows accurate autolanding of planes (by allowing shedding of energy in a similar manner to reverse thrust), but especially gliders, using DIY or commercial airbrake actuators such as [these](http://www.icare-icarus.com/Schambeck-Two-stage-precision-airbrakes-3-6-meter_p_1456.html). An RC channel option is also provided as a manual alternative or override.

This has been extensively tested on board the K1000ULE and other gliders. It has been changed a little to use RCX_OPTION rather than a separate parameter for input channel so I will test it again in the next few days.

### CODE REVIEW

File: ArduPlane/servos.cpp [hunk 1]
```diff
@@ -289,6 +289,46 @@ void Plane::dspoiler_update(void)
...
 
+/*
+ set airbrakes based on reverse thrust and/or manual input RC channel
+ */
+void Plane::airbrake_update(void)
+{
+    // Calculate any manual airbrake input from RC channel option.
+    int8_t manual_airbrake_percent = 0;
+    
+    RC_Channel *airbrake_in_ch = rc().find_channel_for_option(RC_Channel::AUX_FUNC::AIRBRAKE);
```
Comment (tridge) [inline ArduPlane/servos.cpp:300]: find_channel_for_option() is quite expensive, better to do it in set_control_channels() called from one_second_loop().
I notice the flapin channel has the same problem, maybe you could fix that too?

**Review by tridge — CHANGES_REQUESTED** [review]
nice feature! just a small change request

File: ArduPlane/servos.cpp (hunk 1 above)
Comment (magicrub) [reply ArduPlane/servos.cpp:300]: >maybe you could fix that too

good idea for a new PR! Lets get it right for this PR first and then translate it over. As you noticed, this is the same structure as set_servos_flaps()

Comment (magicrub) [reply ArduPlane/servos.cpp:300]: @tridge Are you preferring this PR be blocked until that rework is done?

Comment (tridge) [reply ArduPlane/servos.cpp:300]: yes, it's a tiny change. Just declare airbrake_in_ch in Plane class along with the other channels (roll, pitch etc) and update in set_control_channels()

Comment (peterbarker) [reply ArduPlane/servos.cpp:300]: Could we use an RC channel option instead?

Comment (magicrub) [reply ArduPlane/servos.cpp:300]: @samuelctabor I talked to @tridge and it's just about caching the RC_Channel pointer at 1Hz along with all the other channels in set_control_channels()

**Comment by magicrub** [comment]
LGTM, Thanks for the quick fix @samuelctabor

**Pull request merged**